src/Common/src/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -223,6 +241,12 @@ public override bool ComputeContainsGCPointers(DefType type)
         {
             bool someFieldContainsPointers = false;
 
+            if (type.IsValueType)
```

Comment:
IsGCPointer returns false for `ByReference<T>`, so this not quite self-consistent.

FWIW, the CoreCLR implementation does not treat `ByReference<T>` as GCPointer. The only place where it really matters is the fast path on getClassGClayout - it checks for IsByRefLike to take the slow path there.